### PR TITLE
Fix bug where refactored bootstrap_datasource()

### PR DESCRIPTION
**Pull request by jlrobins**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Unit tests are present
- [x] Have you validated this change locally?

## Summary of Changes

- Fix bug where refactored bootstrap_datasource()  -> connection_registry.factory_and_register() was dropping create_engine_kwargs. Whoops.

## What is the Current Behavior?

- Datasources that needed to pass over create_engine_kwargs / connect_args would fail to be configured properly. That ... didn't include the PostgreSQL I manually clicktested with on Staging.
- No test case covered proving these facts make it down to the create_engine() call.

## What is the New Behavior?

- They will work.
- New test proving that they make it over.

### CODE REVIEW

File: noteable/datasources.py [hunk 1]
```diff
@@ -121,43 +121,43 @@ def bootstrap_datasource(
             metadata['allow_datasource_dialect_autoinstall'],
         )
 
+        # Prepare connection URL string.
```
Comment (jlrobins) [inline noteable/datasources.py:124]: All this goes into the try / catch block so any issues will be captured into add_bootstrapping_failure().

File: noteable/datasources.py [hunk 2]
```diff
@@ -121,43 +121,43 @@ def bootstrap_datasource(
...
+        # according to the needs of this driver?
+        if not metadata['sqlmagic_autocommit']:
+            # A sqlalchemy drivername may be comprised of 'dialect+drivername', such as
+            # 'databricks+connector'.
+            # If so, then we must only pass along the LHS of the '+'.
+            dialect = metadata['drivername'].split('+')[0]
+            add_commit_blacklist_dialect(dialect)
+
+        # Register the connection!
+        connection_registry.factory_and_register(
```
Comment (jlrobins) [inline noteable/datasources.py:145]: The `**create_engine_kwargs` here is the primary bugfix.

File: noteable/sql/connection.py [hunk 3]
```diff
@@ -73,6 +70,7 @@ def __init__(
 
         # SLQA-centric fields hereon down, to be pushed into SQLA subclass in the future.
         self._engine = sqlalchemy.create_engine(connection_url, **create_engine_kwargs)
+        self._create_engine_kwargs = create_engine_kwargs
```
Comment (jlrobins) [inline noteable/sql/connection.py:73]: Stash in the Connection just for use in the test suite as proof they made it here,.

File: noteable/sql/connection.py [hunk 4]
```diff
@@ -133,29 +131,7 @@ def factory_and_register(
...
-                exception=str(e),
-            )
-
-            # Remember the failure so can be shown if / when human tries to use the connection.
-            self.add_bootstrapping_failure(
-                sql_cell_handle=sql_cell_handle, human_name=human_name, error_message=str(e)
-            )
-
-            return
+        conn = self.factory(sql_cell_handle, human_name, connection_url, **kwargs)
```
Comment (jlrobins) [inline noteable/sql/connection.py:134]: Adding to the code in the try/except block in the bootrstapping call then simplified the factory() call not needing to try / catch.

File: tests/test_datasources.py [hunk 5]
```diff
@@ -287,6 +287,7 @@ class SampleData:
             connect_args_dict={
                 'max_download_seconds': '22',
             },
+            expect_identical_connect_args=False,
```
Comment (jlrobins) [inline tests/test_datasources.py:290]: Have to excuse this one, 'cause the SQLite max_download_seconds args is digested entirely by postprocess_sqlite, and won't be seen by the Connection layer.

File: tests/test_datasources.py [hunk 6]
```diff
@@ -413,6 +414,12 @@ def test_bootstrap_datasources(self, datasource_id_factory, tmp_path: Path):
             assert f'@{ds_id}' in registry
             assert sample.meta_dict['name'] in registry
 
+            # Ensure that the Connection actually got bootstrapped with
```
Comment (jlrobins) [inline tests/test_datasources.py:417]: New test(s) over the previously buggy codepath. Sigh.

**Review by kafonek — APPROVED** [review]

**Pull request merged**